AutoRest/Generators/CSharp/CSharp/TemplateModels/ModelTemplateModel.cs

```diff
@@ -187,6 +198,49 @@ private static string CreateSignature(IEnumerable<ConstructorParameterModel> par
                 return string.Join(", ", declarations);
             }
 
+            private static IEnumerable<string> CreateSignatureDocumentation(IEnumerable<ConstructorParameterModel> parameters)
+            {
+                var declarations = new List<string>();
+                IEnumerable<ConstructorParameterModel> parametersWithDocumentation =
+                    parameters.Where(p => !(string.IsNullOrEmpty(p.UnderlyingProperty.Summary) && string.IsNullOrEmpty(p.UnderlyingProperty.Documentation)));
+
+                foreach (var property in parametersWithDocumentation.Where(p => !p.UnderlyingProperty.IsConstant).Select(p => p.UnderlyingProperty))
```

Comment:
Could move the LINQ in here out of the loop and into it's own variable (or pop it up into the other LINQ statement you have above) so it reads like:

```
 IEnumerable<ConstructorParameterModel> parametersWithDocumentation = 
    parameters.Where(p => !(string.IsNullOrEmpty(p.UnderlyingProperty.Summary) &&
    string.IsNullOrEmpty(p.UnderlyingProperty.Documentation)) && !p.UnderlyingProperty.IsConstant).Select(p.UnderlyingProperty);
```

It's minor but it gets all the selection/filtering stuff out of the way early so that you can just do your loop.